### DISCUSSION TITLE
Interactive Metasploit-Like Interface

### DISCUSSION THREAD

**Discussion by TheTechromancer** [blacklanternsecurity/bbot#795] [Ideas]
Courtesy of @TMDeal. Once we have a good way to interact with a database (i.e. both an input and output module for sqlite), it would be cool to have an interactive CLI interface, like recon-ng/metasploit.

**Comment by Sh4d0wHunt3rX** [comment]
I saw conditions in presets, and I thought maybe it's good to have ability to set conditions also when bbot has database. For example, run nuclei only if the link is new (has not been discovered in yesterday scan), or for example do not run paramminer on already discovered links in previous scan ...

**Reply by TheTechromancer** [reply]
@amiremami sometimes I wish BLS employees would pay as much attention to BBOT as you do

**Reply by Sh4d0wHunt3rX** [reply]
Woow man, I really appreciate that 🙏 My interest in bbot was not possible without all of your help to explain me a lot of things even when you are extremely busy. ❤️

**Reply by Sh4d0wHunt3rX** [reply]
Right now, I separated bbot modules in 2 commands. So, first I run passive modules, and then I run aggressive modules. However, there is no way to make bbot understand these are related to one target and merge them. Maybe this is possible when it has database. For example, you have ran a scan on a target without nuclei, but now you want to do nuclei over that result. There should be a feature that make it possible to merge these. Not sure if it's even technically possible.
